.github/workflows/release.yml

```diff
@@ -13,7 +13,7 @@ name: Create a release
 on:
   push:
     branches:
-      - $default-branch
```

Comment:
Can we try ` [ $default-branch ]`? 

This is what the starter workflow users.

Reply:
Also, I think merges might be treated slightly differently, on secondment we used `if: github.ref == 'refs/heads/master'` condition on the steps.

Reply:
That's the same thing isn't it, just different YAML syntax? I don't understand why that would make a difference

Reply:
According to the workflow templates documentation, `$default-branch` is replaced when the workflow is created from the template, so I think that means `$default-branch` has no meaning in a workflow as opposed to a workflow template 

> If you need to refer to a repository's default branch, you can use the $default-branch placeholder. When a workflow is created using your template, the placeholder will be automatically replaced with the name of the repository's default branch.

https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/sharing-workflows-with-your-organization#creating-a-workflow-template

Reply:
fair enough